java/src/jmri/jmrix/powerline/dmx512/SpecificTrafficController.java

```diff
@@ -0,0 +1,153 @@
+            return(true);
+        }
+        return(false);
+    }
+     
+    @Override
+    protected void transmitLoop() {
+        if (oneTimeLog) {
+            oneTimeLog = false;
+            for (int i = 0; i < dmxArray.length; i++) {
+                dmxArray[i] = 0;
+            }
+            dmxArray[0] = (byte) 0; // type of buffer going out
+            /**
+             * deal with thread sync of main tc still getting setup by time of
+             * first call in the transmit loop. Give it time and retry
+             */
+            int tryLimit = 0;
+            while ((activePort == null) && (tryLimit <= 10)) {
+                try {
+                    Thread.sleep(5000);
+                } catch (InterruptedException ignore) {
+                    Thread.currentThread().interrupt();
+                }
+                tryLimit++;
+                activePort = memo.getActiveSerialPort();
+                if (activePort == null) {
+                    log.info("try {} to get activePort", tryLimit);
+                }
+            }
+        }
+
+        // loop forever
+        while (!connectionError && !threadStopRequest) {
+            try {
+                if (ostream != null) {
+                    // break should be for 176 mSec 
+                    if (activePort != null) {
+                        //log.info("Start Break");
+                        activePort.sendBreak(0);
+                        //log.info("Break Sent");
+                    }
+                    /**
+                     * send the buffer of data
+                     */
+                    ostream.write(dmxArray);
+
+                    /**
+                     * wait 25 mSec, then repeat
+                     */
+                    try {
+                        Thread.sleep(25);
+                    } catch (InterruptedException ignore) {
+                        Thread.currentThread().interrupt();
+                    }
+                } else {
+                    connectionError = true;
+                }
+            } catch (IOException | RuntimeException e) {
+                // TODO Currently there's no port recovery if an exception occurs
+                // must restart JMRI to clear xmtException.
+                xmtException = true;
+                portWarn(e);
+            }
+        }
+    }
+
+    // not used, no reback
+    @Override
+    public void receiveLoop() {
+        try {
+            Thread.sleep(10000);
+        } catch (InterruptedException ignore) {
+            Thread.currentThread().interrupt();
+        }
+    }
+    
+    /**
+     * This system provides 256 dim steps
+     */
+    @Override
+    public int getNumberOfIntensitySteps() {
+        return intensitySteps;
+    }
+
+    /**
+     * Send a sequence of Dmx messages
+     * <p>
+     * Makes call to update array
+     */
+    @Override
+    synchronized public boolean sendDmxSequence(int unitid, byte newStep) {
```

Comment:
Why is this method `synchronized`? Synchronized method means that the object is the lock and I don't see any other `synchronized` block on this object. So `synchronized` doesn't seem to protect anything.

Reply:
I think it was carry over from other connection code I started with. Due to the way this works, single common array, you are right it is not needed as sync.